app/server/nbtc/controller.server.ts

```diff
@@ -24,17 +24,26 @@ export default class Controller {
 		this.btcRPCUrl = networkConfig.btcRPCUrl;
 	}
 
-	private async getMintTxs(suiAddr: string): Promise<QueryMintTxResp | Response> {
+	// addr can be sui or btc address
+	private async getMintTxs(addr: string): Promise<QueryMintTxResp | Response> {
 		const method = "nbtc:getMintTxs";
-		if (!isValidSuiAddress(suiAddr)) {
-			return badRequest();
-		}
 		if (!this.btcindexer) {
 			return serverError(method, new Error("Indexer RPC not configured"));
 		}
+
+		let fetchTxsPromise: Promise<NbtcTxResp[]>;
+
+		if (isValidSuiAddress(addr)) {
+			const suiAddr = addr;
+			fetchTxsPromise = this.btcindexer.nbtcMintTxsBySuiAddr(suiAddr);
+		} else {
+			const btcAddr = addr;
+			fetchTxsPromise = this.btcindexer.depositsBySender(btcAddr);
+		}
```

Comment:
this is wrong. If use has connected both wallets we should query both results and merge them.

Solution 2 (maybe even better): 
Rename `depositsBySender` to `nbtcDeposits` with arguments as an object: `{btcAddr?: string, suiAddr?: string})`